python/ucxx/ucxx/_lib/libucxx.pyx

```diff
@@ -489,10 +482,10 @@ cdef class UCXAddress():
         pass
 
     def __reduce__(self) -> tuple:
-        return (UCXAddress.create_from_buffer, (self.string,))
+        return (UCXAddress.create_from_buffer, (bytes(self),))
 
     def __hash__(self) -> int:
-        return hash(bytes(self.string))
+        return hash(bytes(self))
```

Comment:
Once we have `string_view` available in Cython ( https://github.com/cython/cython/pull/6652 ) ( 3.1 ), we may want to revisit this. Then we can write the code to leverage views (fewer copies)